### ISSUE TITLE
The cursor cannot reach the end of a long filename in rename prompt.

### ISSUE THREAD

**Issue by crocket** [ranger/ranger#1780]
#### Runtime Environment

- Operating system and version: Gentoo Linux
- Terminal emulator and version: Alacritty 0.4.0
- Python version: 3.6.9
- Ranger version/commit: 1.9.2
- Locale: en_US.utf8

#### Current Behavior

When I try to rename

```
おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|おはよう|.txt
```

the cursor cannot reach the end of the filename.

#### Expected Behavior

It should be able to reach the end of the filename.

#### Steps to reproduce

1. Create the file.
2. Press `r`
3. Try to reach the end of the filename with a cursor.

**Comment by toonn** [comment]
Hmm, first guess is this is a problem with double-width characters or something. Could you try an equally long filename with latin characters?

**Comment by crocket** [comment]
With a long filename in english alphabet, the issue doesn't exist.

**Comment by toonn** [comment]
And if you go to the end of the filename and then delete 5 chars, for example, does the resulting filename reflect that, i.e., not what is shown but what `ls` will show after executing the rename?

**Comment by crocket** [comment]
The cursor actually goes to the end and deletes 5 characters, but it is not rendered on my terminal.

**Comment by toonn** [comment]
I think there's nothing we can do about this unfortunately. It's a shortcoming of the terminal's rendering of double-width characters. If you know of a python program using ncurses that does this right that'd be a great help. Maybe there *is* something we can do.

In the meantime I only have a stopgap solution to offer which is to use `:bulkrename` on such files (even if it's just one file), I assume vim or most any editor you use would handle these characters better.

**Comment by crocket** [comment]
Yes, I use bulkrename for now.

**Comment by crocket** [comment]
I tested this on xfce4-terminal, and the issue disappears.
I think this issue should be moved to alacritty.

**Comment by crocket** [comment]
I reproduced this issue on xfce4-terminal, too.

**Comment by toonn** [comment]
Yeah, I don't know of any terminals or programs that do this properly. Except for editors like vim and emacs. So I guess we'd need to look there.

**Comment by qrvd** [comment]
I think it's a problem with the Console class. It seems to mix up the concept of *character index* with that of *screen index* so when wide characters are involved it miscalculates both the cursor position and how far the console line needs to be offset before having ncurses draw it onto the screen. It took a while to figure out but I've now pushed a branch that seems to properly fix this.

**Comment by hut** [comment]
Possibly this is fixed by https://github.com/ranger/ranger/pull/1982 or https://github.com/ranger/ranger/pull/2481 (or not.^^")
